src/coinbase/validator.ts

```diff
@@ -151,11 +156,119 @@ export class Validator {
+    return this.model.details?.activationEpoch || "";
+  }
+
+  /**
+   * Returns the balance of the validator.
+   *
+   * @returns The balance object.
+   */
+  public getBalance(): Balance | undefined {
+    return this.model.details?.balance;
+  }
+
+  /**
+   * Returns the effective balance of the validator.
+   *
+   * @returns The effective balance object.
+   */
+  public getEffectiveBalance(): Balance | undefined {
+    return this.model.details?.effective_balance;
+  }
+
+  /**
+   * Returns the exit epoch of the validator.
+   *
+   * @returns The exit epoch as a string.
+   */
+  public getExitEpoch(): string {
+    return this.model.details?.exitEpoch || "";
+  }
+
+  /**
+   * Returns the index of the validator.
+   *
+   * @returns The validator index as a string.
+   */
+  public getIndex(): string {
+    return this.model.details?.index || "";
+  }
+
+  /**
+   * Returns the public key of the validator.
+   *
+   * @returns The validator's public key as a string.
+   */
+  public getPublicKey(): string {
+    return this.model.details?.public_key || "";
+  }
+
+  /**
+   * Returns whether the validator has been slashed.
+   *
+   * @returns True if the validator has been slashed, false otherwise.
+   */
+  public isSlashed(): boolean {
+    return this.model.details?.slashed || false;
+  }
+
+  /**
+   * Returns the withdrawable epoch of the validator.
+   *
+   * @returns The withdrawable epoch as a string.
+   */
+  public getWithdrawableEpoch(): string {
+    return this.model.details?.withdrawableEpoch || "";
+  }
+
+  /**
+   * Returns the withdrawal address of the validator.
+   *
+   * @returns The withdrawal address as a string.
+   */
+  public getWithdrawalAddress(): string {
+    return this.model.details?.withdrawal_address || "";
+  }
+
+  /**
+   * Returns the string representation of the Validator including all its details.
+   *
+   * @returns The string representation of the Validator including all its details.
    */
   public toString(): string {
-    return `Id: ${this.getValidatorId()} Status: ${this.getStatus()}`;
+    return `Validator { Id: ${this.getValidatorId()}, Status: ${this.getStatus()}, Exit Epoch: ${this.getExitEpoch()}, Network ID: ${this.getNetworkId()}, Asset ID: ${this.getAssetId()}, Index: ${this.getIndex()}, Public Key: ${this.getPublicKey()}, Slashed: ${this.isSlashed()}, Withdrawable Epoch: ${this.getWithdrawableEpoch()}, Withdrawal Address: ${this.getWithdrawalAddress()}, Effective Balance: { Amount: ${this.getEffectiveBalance()?.amount}, AssetID: { ${this.getEffectiveBalance()?.asset.asset_id}, Decimals: ${this.getEffectiveBalance()?.asset.decimals}, NetworkID: ${this.getEffectiveBalance()?.asset.network_id} } }, Balance: { Amount ${this.getBalance()?.amount}, Asset: { AssetID: ${this.getBalance()?.asset.asset_id}, Decimals: ${this.getBalance()?.asset.decimals}, NetworkID: ${this.getBalance()?.asset.network_id} } }, Activation Epoch: ${this.getActivationEpoch()} }`;
```

Comment:
I think toString should still be the simpler one. Primarily used of logging etc. This will be too much to log.

Reply:
Yeah I see. Truly wasn't sure what to do here. I'm fine with simplifying this.
Are you suggesting just printing out the top level data values? or what subset specifically would be important?

Reply:
Just id and status like before.